src/global/reco/GeneratedJets_factory.cc

```diff
@@ -61,24 +61,19 @@ namespace eicrecon {
         // grab input collection
         auto input = static_cast<const edm4hep::MCParticleCollection*>(event->GetCollectionBase(GetInputTags()[0]));
 
-        // extract particle momenta
-        std::vector<const edm4hep::LorentzVectorE*> momenta;
+        // select only final state particles for reconstruction
+        // TODO: Need to exclude the scattered electron
+        edm4hep::MCParticleCollection for_reconstruction;
+        for_reconstruction.setSubsetCollection(true);
         for (const auto& particle : *input) {
-
-            // select only final state charged particles
             const bool is_final_state = (particle.getGeneratorStatus() == 1);
-            if (!is_final_state) continue;
-
-            const auto& momentum = particle.getMomentum();
-            const auto& energy = particle.getEnergy();
-            momenta.push_back(new edm4hep::LorentzVectorE(momentum.x, momentum.y, momentum.z, energy));
-        }  // end particle loop
+            if (is_final_state) {
```

Comment:
Thanks for simplifying the factory for ReconstructedJets. But can we stick this here in the GeneratedJets algorithm too so the GeneratedJets factory is minimalistic again? I was just looking into converting these to JOmniFactory for better external wiring, and that's the main reason we would not be able to do this: the factory is too specific and we want it to be generic.

Reply:
That makes sense! Truth be told, I think it would be very nice to use 1 generic factory for both the `GeneratedJets` and `ReconstructedJets`. Off the top of my head, I can think of 2 ways of doing this:

1. the 1st would be to move the `is_final_state` selection to the algorithm and set it up so that it only checks for this if the input is an `edm4hep::MCParticle` collection;

2. and the 2nd would be to make a "final state" selector (like the `ChargedParticleSelector` factory) that picks out only final state particles and saves them to a collection which would then be the input for the `GeneratedJets`....

In the case of the former, it might make sense to make the desired generator status a user-configurable parameter.

Reply:
Instead of attaching to the MCParticles collection (of type MCParticleCollection) in GeneratedJets_factory, why not attach to the GeneratedParticles collection (of type ReconstructedParticleCollection) from the MC2SmearedParticle factory? Then both GeneratedJets and ReconstructedJets can both work on ReconstructedParticle collections, instead of having different types.

Reply:
Ah! Using the `GeneratedParticles` collection would also render the issue with picking out only final state particles moot, right? That would let us move to all PODIO relations and vastly simplify the jet factory in one go...

That being said, I do agree with @veprbl that it would be nice to exercise the templated code. But it might be more prudent for us to focus on getting the Jet EDM in and come back to it afterwards...